### ISSUE TITLE
Plugin fails on Gradle 2.14

### ISSUE THREAD

**Issue by IRus** [asciidoctor/asciidoctor-gradle-plugin#195]
My configuration file:

```
plugins {
    id "org.asciidoctor.convert" version "1.5.3"
}

asciidoctorj {
    version = '1.5.4'
}

asciidoctor {
    sources {
        include "index.adoc"
        include "/images/"
    }
    sourceDir 'docs'
    outputDir 'docs-out'
    attributes 'toc': 'left',
               'toc-title': 'Table of Contents'
}
```

After upgrade to Gradle 2.14 i got following error, when running `asciidoctor` task:

```
11:47:42 AM: Executing external task 'asciidoctor'...
(ArgumentError) asciidoctor: FAILED: /home/joda/dev/project/projectname/project-documentation/docs/images/checkstyle.png: Failed to load AsciiDoc document - invalid byte sequence in UTF-8
:project-documentation:asciidoctor FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':project-documentation:asciidoctor'.
> Error running Asciidoctor

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 10.133 secs
(ArgumentError) asciidoctor: FAILED: /home/joda/dev/project/projectname/project-documentation/docs/images/checkstyle.png: Failed to load AsciiDoc document - invalid byte sequence in UTF-8
```

**Comment by mojavelinux** [comment]
You are asking the plugin to run the converter on images, hence the invalid byte sequence error. Drop the `include "/images/"` line. Sources should only be AsciiDoc files.

It does make me wonder if perhaps we should be checking for this sort of thing. I don't see why the plugin should attempt to run on any file that doesn't have an AsciiDoc file extension.

**Comment by IRus** [comment]
Hm, i doesn't remember why i added '/images/' to sources(i wrote this config like year ago), but your suggestion works, thank you!

**Comment by abelsromero** [comment]
> It does make me wonder if perhaps we should be checking for this sort of thing. I don't see why the plugin should attempt to run on any file that doesn't have an AsciiDoc file extension.

Current logic takes ALL files that match the patterns in `sources` block ([ref](https://github.com/asciidoctor/asciidoctor-gradle-plugin/blob/master/src/main/groovy/org/asciidoctor/gradle/AsciidoctorTask.groovy#L470)). Only if no sources block is set, it filters all valid AsciiDoc files found in `sourceDir` by file extension. This creates some ambiguity in the sense that makes possible to render more than one folder using different options: more than one task using sourceDir, one task with sources (which need to add the file extension like 'other-docs/**/*.adoc'), etc.

I feel we are struggling with the same issue as in the maven plugin, given some POCs I would avoid adding more attributes, like validFileExtensions. I don't know if it's possible, but ideally I'd like to use a single `sources` block instead of the combo `sourceDir + sources`. If the include is a folder, I'd like that the plugin takes only the AsciiDoc documents. If I use a custom extension, then I'll add as a pattern.

Btw, the reason why this fails in 2.14 and not previous versions is because the / at the beginning of `/images/`. With the last Gradle version, this resolves relative to the `sourceDir`...not sure if this a bug or it is intended. If you use `images/` with previous versions it fails with the same error.

**Comment by aalmiray** [comment]
Is this still an issue? I hope most users have moved to Gradle 3.x by now

**Comment by IRus** [comment]
I solved this problem in 2.14.

**Comment by aalmiray** [comment]
OK. Closing. Thank you for the prompt reply.
